src/gov-action-contracts/AIPs/upgrade-executor-upgrade/UpgradeExecutorUpgradeAction.sol

```diff
@@ -0,0 +1,17 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.16;
+
+import {UpgradeExecutor} from "@offchainlabs/upgrade-executor/src/UpgradeExecutor.sol";
+import {ERC1967Upgrade} from "@openzeppelin/contracts/proxy/ERC1967/ERC1967Upgrade.sol";
+
+contract UpgradeExecutorUpgradeAction is ERC1967Upgrade {
+    address public immutable newUpgradeExecutorImplementation = address(new UpgradeExecutor());
+
+    function perform() external {
+        _upgradeTo(newUpgradeExecutorImplementation);
```

Comment:
not a big fan of using `ERC1967Upgrade` to modify the storage slot directly, can we just have it go through the proxy admin instead?

Reply:
yeah we can do that, i have a couple versions at earlier commits that do

eg https://github.com/ArbitrumFoundation/governance/pull/319/commits/d74a6d9373506202b42d4e3bdbff58d1fe0d6bfa#diff-96112336c8e4f94e4f509a6e67874a61d2637890f110f1b10e95998a50996812

eg2 https://github.com/ArbitrumFoundation/governance/pull/319/commits/90fb0bc329c32656af3843d5f9ee7746da79b64d#diff-96112336c8e4f94e4f509a6e67874a61d2637890f110f1b10e95998a50996812

Reply:
reverted back to the first example